contracts/claimManager.sol

```diff
@@ -0,0 +1,242 @@
+
+        claim.bondedDeposits[account] = claim.bondedDeposits[account].add(amount);
+        DepositBonded(claimID, account, amount);
+        return claim.bondedDeposits[account];
+    }
+
+    // @dev – accessor for a claims bonded deposits.
+    // @param claimID – the claim id.
+    // @param account – the user's address.
+    // @return – the user's deposit bonded for the claim.
+    function getBondedDeposit(uint claimID, address account) public view returns (uint) {
+        ScryptClaim storage claim = claims[claimID];
+        return claim.bondedDeposits[account];
+    }
+
+    // @dev – unlocks a user's bonded deposits from a claim.
+    // @param claimID – the claim id.
+    // @param account – the user's address.
+    // @return – the user's deposit which was unbonded from the claim.
+    function unbondDeposit(uint claimID, address account) private returns (uint) {
+        ScryptClaim storage claim = claims[claimID];
+        uint bondedDeposit = claim.bondedDeposits[account];
+        delete claim.bondedDeposits[account];
+        deposits[account] = deposits[account].add(bondedDeposit);
+        DepositUnbonded(claimID, account, bondedDeposit);
+
+        return bondedDeposit;
+    }
+
+    // @dev – check whether a DogeCoin blockHash was calculated correctly from the plaintext block header.
+    // only callable by the DogeRelay contract.
+    // @param _plaintext – the plaintext blockHeader.
+    // @param _blockHash – the blockHash.
+    // @param claimant – the address of the Dogecoin block submitter.
+    function checkScrypt(bytes _plaintext, bytes _blockHash, address claimant) onlyBy(dogeRelayAddress) public returns (uint) {
+        require(deposits[claimant] >= minDeposit);
+
+        ScryptClaim storage claim = claims[numClaims];
+        claim.claimant = claimant;
+        claim.plaintext = _plaintext;
+        claim.blockHash = _blockHash;
+        claim.numChallengers = 0;
+        claim.currentChallenger = 0;
+        claim.verificationOngoing = false;
+        claim.createdAt = block.number;
+
+        bondDeposit(numClaims, claimant, minDeposit);
+        ClaimCreated(numClaims, claim.claimant, claim.plaintext, claim.blockHash);
+
+        numClaims.add(1);
+        return numClaims;
+    }
+
+    // @dev – challenge an existing Scrypt claim.
+    // triggers a downstream claim computation on the scryptVerifier contract
+    // where the claimant & the challenger will immediately begin playing a verification.
+    //
+    // @param claimID – the claim ID.
+    function challengeClaim(uint claimID) public {
+        ScryptClaim storage claim = claims[claimID];
+
+        // check that the claim is in Stage 1 (i.e. accepting new challenges).
+        require(block.number.sub(claim.createdAt) <= challengeTimeout);
+
+        require(deposits[msg.sender] >= minDeposit);
+        bondDeposit(claimID, msg.sender, minDeposit);
+
+        claim.challengers[claim.numChallengers] = msg.sender;
```

Comment:
Convert to array?

Reply:
Created an issue: https://github.com/TrueBitFoundation/scrypt-interactive/issues/39